packages/firestore/src/api/user_data_writer.ts

```diff
@@ -75,10 +75,11 @@ export class UserDataWriter<T = firestore.DocumentData> {
       case TypeOrder.RefValue:
         return this.convertReference(value.referenceValue!);
       case TypeOrder.GeoPointValue:
-        return new GeoPoint(
-          value.geoPointValue!.latitude!,
-          value.geoPointValue!.longitude!
-        );
+        // `geoPointValue` stores zeroes as undefined, which will fail
```

Comment:
Sebastian, can you think of any other types that might be affected? `Timestamp`, I think, isn't.

Reply:
`Timestamp` goes through `normalizeNumber()` which turns `undefined` into `0`. Do you want to follow this pattern here as well and invoke `normalizeNumber()`? It also correctly handles NaN and Infinity, which should never be part of a GeoPoint, but who knows where 2020 is going to take us.

Reply:
Done (and extracted into a small helper function for consistency with `Timestamp` et al.).